### ISSUE TITLE
Data transformations in the 'income prediction' example are performed on the entire dataset, rather than just on the train one

### ISSUE THREAD

**Issue by MarcoGorelli** [SeldonIO/alibi#140]
The `anchor_text_movie` example correctly fits `CountVectorizer(min_df=1)` to the train set and then uses it to transform both train and test sets.

In the `anchor_tabular_adult` example, on the other hand, we have
```
preprocessor = ColumnTransformer(transformers=[('num', ordinal_transformer, ordinal_features),
                                               ('cat', categorical_transformer, categorical_features)])
preprocessor.fit(data)
```

and then 

```
clf.fit(preprocessor.transform(X_train), Y_train)
predict_fn = lambda x: clf.predict(preprocessor.transform(x))
```
.

In `Hands-On Machine Learning with Scikit-Learn & TensorFlow`, it says

> As with all the transformations, it is important to fit the scalers to the training data only, not to the full dataset (including the test set). Only then can you use them to transform the training set and the test set (and new data)

Although it hardly makes any difference here, it's probably a better practice to fit the preprocessing to the train set rather than to the entire dataset.

**Comment by jklaise** [comment]
Good catch, thanks!
